lib/elixir/lib/calendar/duration.ex

```diff
@@ -197,4 +197,48 @@ defmodule Duration do
       microsecond: {-ms, p}
     }
   end
+
+  @doc """
+  Parses an ISO 8601-2 formatted duration string to a `Duration` struct.
```

Comment:
```suggestion
  Parses an ISO 8601 formatted duration string to a `Duration` struct.
```

We don't precisely support ISO8601, because we don't support fractional parts everywhere. It would be better to be explicit and say:

1. We only support fractions (using `.`) in the seconds unit
2. We extend the format to support negating individial units
3. We support weeks between days and months (probably with an example)